src/xmake.ts

```diff
@@ -648,106 +641,34 @@ export class XMake implements vscode.Disposable {
                 if (result) {
                     targetName = result.split('__end__')[0].trim();
                 }
             }
         }
 
-        // get target arguments
-        let args = ["run"];
-        let configArgs = [];
-        if (targetName && targetName in config.runningTargetsArguments)
-            configArgs = config.runningTargetsArguments[targetName];
-        else if ("default" in config.runningTargetsArguments)
-            configArgs = config.runningTargetsArguments["default"];
-
-        // make command
-        let command = config.executable;
-        if (targetName && targetName != "default") {
-            args.push(targetName);
-            for (let arg of configArgs) {
-                args.push(arg);
-            }
-        } else if (targetName == "all") {
-            args.push("-a");
-        } else {
-            for (let arg of configArgs) {
-                args.push(arg);
-            }
-        }
-
-        // build and run it
-        await this.onBuild(target);
-        await this._terminal.execv("run", command, args);
-    }
-
-    // on run target
-    async onRun(target?: string) {
-
-        // this plugin enabled?
-        if (!this._enabled) {
-            return
-        }
-
-        // build and run?
         const runMode = config.get<string>("runMode");
         if (runMode == "buildRun") {
-            return this.onBuildRun(target)
-        }
-
-        // get target name
-        let targetName = this._option.get<string>("target");
-        if (!targetName) {
-            let getDefaultTargetPathScript = path.join(__dirname, `../../assets/default_target.lua`);
-            if (fs.existsSync(getDefaultTargetPathScript)) {
-                let result = (await process.iorunv(config.executable, ["l", getDefaultTargetPathScript], {"COLORTERM": "nocolor"}, config.workingDirectory)).stdout.trim();
-                if (result) {
-                    targetName = result.split('__end__')[0].trim();
-                }
-            }
-        }
-
-        // get target arguments
-        let args = ["run"];
-        let configArgs = [];
-        if (targetName && targetName in config.runningTargetsArguments)
-            configArgs = config.runningTargetsArguments[targetName];
-        else if ("default" in config.runningTargetsArguments)
-            configArgs = config.runningTargetsArguments["default"];
-
-        // make command
-        let command = config.executable;
-        if (targetName && targetName != "default") {
-            args.push(targetName);
-            for (let arg of configArgs) {
-                args.push(arg);
-            }
-        } else if (targetName == "all") {
-            args.push("-a");
-        } else {
-            for (let arg of configArgs) {
-                args.push(arg);
-            }
```

Comment:
oh, why?

Reply:
It can be configured with args in launch.json